src/token/WstEthL2Token.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity >=0.8.8 <0.9.0;
+
+import {OwnableUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {ERC20Upgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/token/ERC20/ERC20Upgradeable.sol";
+import {ERC20BurnableUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
+import {UUPSUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {INttToken} from "@wormhole-foundation/native_token_transfer/interfaces/INttToken.sol";
+
+/// @title WstEthL2Token
+/// @notice A UUPS upgradeable token with access controlled minting and burning.
+contract WstEthL2Token is
+    INttToken,
+    UUPSUpgradeable,
+    ERC20Upgradeable,
+    ERC20BurnableUpgradeable,
+    OwnableUpgradeable
+{
+    // =============== Storage ==============================================================
+
+    struct MinterStorage {
+        address _minter;
+    }
+
+    bytes32 private constant MINTER_SLOT = bytes32(uint256(keccak256("wsteth.minter")) - 1);
+
+    // =============== Storage Getters/Setters ==============================================
+
+    function _getMinterStorage() internal pure returns (MinterStorage storage $) {
+        uint256 slot = uint256(MINTER_SLOT);
+        assembly ("memory-safe") {
+            $.slot := slot
+        }
+    }
+
+    /// @notice A function to set the new minter for the tokens.
+    /// @param newMinter The address to add as both a minter and burner.
+    function setMinter(address newMinter) external onlyOwner {
+        _getMinterStorage()._minter = newMinter;
+    }
+
+    /// @dev Returns the address of the current minter.
+    function minter() public view virtual returns (address) {
+        MinterStorage storage $ = _getMinterStorage();
+        return $._minter;
+    }
+
+    /// @dev Throws if called by any account other than the minter.
+    modifier onlyMinter() {
+        if (minter() != _msgSender()) {
+            revert UnauthorizedAccount(_msgSender());
```

Comment:
Use `CallerNotMinter` error from the NTT interface instead of the new `UnauthorizedAccount` error?